rspec-expectations/lib/rspec/matchers/built_in/base_matcher.rb

```diff
@@ -137,8 +137,18 @@ module HashFormatting
           #     {:a => 5, :b => 2}
           #
           # This is idempotent and safe to run on a string multiple times.
+          # On Ruby 3.4 and later, it also adds spaces around the hash rocket
+          # and converts `a: 1` to `:a => 1` for consistency.
           def improve_hash_formatting(inspect_string)
-            inspect_string.gsub(/(\S)=>(\S)/, '\1 => \2')
+            if RUBY_VERSION.to_f >= 3.4
+              if inspect_string.match?(/(\w+):/)
```

Comment:
This is for me quite ugly. :/  I am not very confident with this change.

We have to do this in order to stay as much as possible compatible with previous RSpec output. But the thing is we are going the opposite way of the new hash output in Ruby 3.4.

I am wondering if we should be more flexible on the spec output in our reporting and simply have helper to match in our own specs the 3 formats:
- `{:a=>1}`
- `{:a => 1}`
- `{a: 1}`

Maybe something to change in dedicated version ?